climpred/classes.py

```diff
@@ -819,164 +822,111 @@ def verify(
-                - maximize/None: maximize the degrees of freedom by slicing ``hind`` and
-                ``verif`` to a common time frame at each lead.
-                - same_inits: slice to a common init frame prior to computing
-                metric. This philosophy follows the thought that each lead should be
-                based on the same set of initializations.
-                - same_verif: slice to a common/consistent verification time frame prior
-                to computing metric. This philosophy follows the thought that each lead
-                should be based on the same set of verification dates.
-
-        Returns:
-            Dataset of comparison results (if comparing to one observational product),
-            or dictionary of Datasets with keys corresponding to observations short
-            name.
-        """
-        warnings.warn(
-            '`HindcastEnsemble.compute_metric()` will be deprecated. '
-            'Using `HindcastEnsemble.verify()` is encouraged.',
-            PendingDeprecationWarning,
-        )
-        return self.verify(
-            name=name, metric=metric, comparison=comparison, alignment=alignment
-        )
-
-    def compute_uninitialized(self, name=None, metric='pearson_r', comparison='e2o'):
-        """Verifies the uninitialized ensemble against observations/verification data.
-
-        This will automatically verify against all shared variables
-        between the uninitialized ensemble and observations/verification data.
 
-        Args:
-            name (str): Short name of observations/verification data to compare to.
-                If ``None``, compare to all observations/verification data.
-            metric (str, default 'pearson_r'): Metric to apply for verification.
-            comparison (str, default 'e2o'): How to compare to the
-                observations/verification data. ('e2o' for ensemble mean to
-                observations/verification data. 'm2o' for each individual member to
-                observations/verification data).
-            alignment (str): which inits or verification times should be aligned?
-                - maximize/None: maximize the degrees of freedom by slicing ``hind`` and
-                ``verif`` to a common time frame at each lead.
-                - same_inits: slice to a common init frame prior to computing
-                metric. This philosophy follows the thought that each lead should be
-                based on the same set of initializations.
-                - same_verif: slice to a common/consistent verification time frame prior
-                to computing metric. This philosophy follows the thought that each lead
-                should be based on the same set of verification dates.
+        def _verify(
+            hind,
+            verif,
+            hist,
+            reference,
+            metric,
+            comparison,
+            alignment,
+            dim,
+            **metric_kwargs,
```

Comment:
Maybe in the next major release of climpred, we could create internal classes so that we don't have to pass in this many arguments for every function and just use self.metric, self.comparison, etc.

Reply:
Yeah I was thinking about how to clean something like this up. So `metric` and `comparison` are currently classes (see `metrics.py` and `comparisons.py`). Maybe you mean having them as methods or attributes to our classes here?